### ISSUE TITLE
[Fargate] [request]: Windows support

### ISSUE THREAD

**Issue by Vlaaaaaaad** [aws/containers-roadmap#508]
**Tell us about your request**
Windows support for Fargate would be awesome.

[EKS has beta support for Windows nodes](https://github.com/aws/containers-roadmap/issues/69) and I imagine Fargate support might be relatively straightforward( well, as straightforward as things can be in software which is rather... *bendy*). But Fargate for EKS [seems delayed](https://github.com/aws/containers-roadmap/issues/32) so Fargate for ECS might be the only option.

**Which service(s) is this request for?**
Fargate for ECS/ EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Some specific usecases/ features **need** Windows due to Windows-only software/ licenses. For some of them Wine can be used to get the workload running happily-ish in Linux Docker containers on EKS. For others an ASG/ EC2 instance with Windows needs to be created and maintained.

Many of the usecases are temporary/ on-demand and I'd love the option to have an app in EKS crate a Fargate task for processing the Windows workload.

**Are you currently working around this issue?**
EKS with Windows nodes can be used as a workaround.

**Comment by jpoley** [comment]
This would be great +1 for Fargate supporting windows containers.  It’s not trivial, but it will really helpful.

**Comment by rahulguptaadobe** [comment]
For serious adoption, especially for Windows support, clients would rather be managing their containers than how and what is being managed on the nodes, in terms of windows version, other privileged and non privileged containers being managed to maintain compatibility with control plane for ingress and CNI. For advanced users, GPU support on those nodes and how the devices are available to running client containers should also be standardized as commodity offering.

**Comment by Type1J** [comment]
I would like Windows support as well, but I think until the kernel version is decoupled with container images, I'm not sure if this will happen.

**Comment by halovanic** [comment]
Windows used to require strictly matching patch revisions between host and container, but that has been lifted [as of 1809 / Windows Server 2019](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-20H2%2Cwindows-10-20H2#revision-number-patching).

**Comment by Type1J** [comment]
Thanks! That's good to know. As soon as I get a chance to verify, I have a few clusters that I need to update.

**Comment by zzFernando** [comment]
Hi guys, the fargate support for windows is coming soon ?

**Comment by Type1J** [comment]
@zzFernando: It doesn't seem so. If you want Windows on ECS or EKS, you must use the EC2 launchtype, for now.

**Comment by mreferre** [comment]
Let's say it sits between "working on it" and "coming soon". Windows brings in its own peculiarities and we are still working on some of the details.

**Comment by Vlaaaaaaad** [comment]
Since this request is pretty popular, I updated the top comment with the latest links for Windows support (EKS support moved from _beta_ to _stable_ since I created this issue, for example).

@mreferre was faster than me, but, building on that, I also want to highlight this part from the FAQ on this repository:
> Q: What do the roadmap categories mean?
>
> Just shipped - obvious, right?
> Coming soon - coming up. Think **a couple of months out, give or take.**
> We're working on it - **in progress, but further out.** We might still be working through the implementation details, or scoping stuff out.
> Researching - We're thinking about it. This might mean we're still designing, or thinking through how this might work. This is a great phase to send how you want to see something implemented! We'd love to see your usecase or design ideas here.

Barring any dramatic Windows behavior, I imagine we'll see an initial release sometime during the winter.

**Comment by mreferre** [comment]
@lewislabs yes. Actually that is the only mode supported by Fargate because.... there are not hosts.

**Comment by h3smith** [comment]
Thank you all for moving this forward. This will be a huge win for us. We hate managing ECS clusters. Fargate automation will be a huge shift.

**Comment by james-skinner-deltatre** [comment]
Great news this has shipped 👍🏼. Would love to migrate from EC2 hosts...is there any plan to add support for other windows versions? Specifically 2022.

**Comment by SaloniSonpal** [comment]
@grochoge - we are considering EKS/Fargate support based on customer demand, so I have opened #1546 to track it. Please upvote and help us prioritize.

**Comment by SaloniSonpal** [comment]
@james-skinner-deltatre - yes, we are planning on adding support for newer Windows Server versions, now tracking with #1547

**Comment by SwagMuffinMcYoloPants** [comment]
Will GovCloud eventually support this feature?

**Comment by SwagMuffinMcYoloPants** [comment]
@SaloniSonpal , Is this still on the roadmap for govcloud support?
